src/filters/authfilter.bal

```diff
@@ -0,0 +1,186 @@
+import ballerina/config;
+import ballerina/encoding;
+import ballerina/http;
+import ballerina/log;
+import ballerina/stringutils;
+
+
+
+public type AuthFilter object {
+
+    private http:Client clientEndpoint;
+
+    public function __init() {
+        self.clientEndpoint = new (config:getAsString("mosip.auth.base_url"));
+    }
+    *http:RequestFilter;
+
+    public function filterRequest(http:Caller caller, http:Request request,
+        http:FilterContext context) returns boolean {
+        if (config:getAsString("mosip.auth.filter_status", "disable") == "enable") {
```

Comment:
This method has too many nested conditions and method is very big, please revisit the code see if we can avoid nesting and break it into multiple methods to make the code more maintainable.

Eg. 1. Check for the negative case first in If condition and return from there to avoid nesting
2. publish and subscribe handling make it to 2 different methods